### PR TITLE
firefox: use nss without p11-kit

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

Quickfix to allow firefox to recognize certificates as trusted by
Mozilla.

Downside: NixOS managed certificates will not be passed into firefox anymore until we fixup cacert.

Related: #126065

CC: @rnhmjoj @NixOS/nixos-release-managers 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dasJ** [comment]
Does this also need a backport to 20.09?

**Comment by andir** [comment]
Don't care about Firefox. Do as you see fit.

File: pkgs/applications/networking/browsers/firefox/common.nix [hunk 1]
```diff
@@ -116,7 +116,7 @@ let
                 then overrideCC stdenv llvmPackages.clangUseLLVM
                 else stdenv;
 
-  nss_pkg = if lib.versionOlder ffversion "83" then nss_3_53 else nss;
+  nss_pkg = if lib.versionOlder ffversion "83" then nss_3_53 else nss.override { useP11kit = false; };
```
Comment (jonringer) [inline pkgs/applications/networking/browsers/firefox/common.nix:119]: do you mind adding a comment, for future contributors as well as seems unusual

Comment (mweinelt) [reply pkgs/applications/networking/browsers/firefox/common.nix:121]: Yes, done.

**Review by jonringer — APPROVED** [review]
LGTM

**Comment by rnhmjoj** [comment]
Can you also comment out firefox from `nixos/tests/custom-ca.nix`? Otherwise the test will break.

**Comment by mweinelt** [comment]
@GrahamcOfBorg test custom-ca

**Comment by mweinelt** [comment]
Tested on x86_64-linux, built on aarch64-linux. The test continues to request screendumps.

**Review by ajs124 — APPROVED** [review]

**Pull request merged**